06.10.2025/sortedList.c

```diff
@@ -0,0 +1,145 @@
+#include "sortedList.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct ListNode {
+    int value;
+    struct ListNode* next;
+} ListNode;
+
+static ListNode* allocListNode(int value);
+
+ListNode* allocListNode(int value)
+{
+    ListNode* ptr = malloc(sizeof(struct ListNode));
+    if (ptr == NULL)
+        return ptr;
+
+    ptr->value = value;
+    ptr->next = NULL;
+    return ptr;
+}
+
+void sortedListInit(SortedList* pList)
+{
+    pList->_head = NULL;
+}
+
+void sortedListFree(SortedList* pList)
+{
+    ListNode* ptr = pList->_head;
+    while (ptr != NULL) {
+        ListNode* temp = ptr->next;
+        free(ptr);
+        ptr = temp;
+    }
+}
+
+bool sortedListInsert(SortedList* pList, int value)
+{
+    ListNode* newNode = allocListNode(value);
+    if (newNode == NULL)
+        return false;
+
+    if (pList->_head == NULL) {
+        pList->_head = newNode;
+        return true;
+    } else if (pList->_head->value >= value) {
```

Comment:
```suggestion
    }
    
    if (pList->_head->value >= value) {
```

Так было бы понятнее, что мы сначала разибраем крайние случаи, а потом вставляем "на общих основаниях"